repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -206,6 +222,74 @@ object Evaluator{
       )
     }
 
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
+      _ <- Catching{ case e: ThreadDeath => interrupted() }
+      (classFiles, importData) <- compileClass(
+        s"""
```

Comment:
Could we extract this and the similar code above in `processLine` into a common function to make sure it stays in sync?